src/lib/private-key-connector/index.ts

```diff
@@ -0,0 +1,227 @@
+/* eslint-disable max-classes-per-file */
+/* eslint-disable no-prototype-builtins */
+/* eslint-disable no-prototype-builtins */
+/* eslint-disable no-prototype-builtins */
+import { AbstractConnectorArguments, ConnectorUpdate } from "@web3-react/types";
+import { AbstractConnector } from "@web3-react/abstract-connector";
+import warning from "tiny-warning";
+
+import { SendReturnResult, SendReturn, Send, SendOld, Ethereum } from "./types";
+
+declare global {
+  interface Window {
+    ethereum?: Ethereum;
+  }
+}
+
+declare const __DEV__: boolean;
+
+function parseSendReturn(sendReturn: SendReturnResult | SendReturn): any {
+  return sendReturn.hasOwnProperty("result") ? sendReturn.result : sendReturn;
+}
+
+export class NoEthereumProviderError extends Error {
+  public constructor() {
+    super();
+    this.name = this.constructor.name;
+    this.message = "No Ethereum provider was found on window.ethereum.";
+  }
+}
+
+export class UserRejectedRequestError extends Error {
+  public constructor() {
+    super();
+    this.name = this.constructor.name;
+    this.message = "The user rejected the request.";
+  }
+}
+
+export class PrivateKeyConnector extends AbstractConnector {
```

Comment:
Is it not possible to use `ethers.Wallet` in combination with `ethers.provider.InfuraProvider` or something like this, rather than having to implement a connector ourselves?

Reply:
I think there probably is a better way to do this that doesn't require a custom connector. I spend some time trying options including `InfuraProvider` but wasn't able to get anything working unfortunately.

The main challenge that I encountered was that we are using the `useWeb3React` hook for lots of stuff, and for that to activate we either need to use one of their connectors, or implement our own one. I tried using their `WalletConnect` connector but the API it expected for that connector was different to anything I was able to generate using `InfuraProvider` or `PrivateKeyProvider` ect.

I'm probably missing something though! If you have any suggestions on how we might be able to link up a provider with `useWeb3React` more easily would be great!! 😃